src/index.js

```diff
@@ -1,7 +1,349 @@
+    let userName = evt.target.value;
+    this.setState({
+      userName: userName,
+      
+    })
+  }
+
+
+  //This function allows user to make a guess. Fetches the county, if the guess is the same county displays score. If guess is incorrect, subtracts ten from score and displays score
+  makeGuess = (evt) => {
+    evt.preventDefault()
+
+    let selectedCounty = document.getElementById("countyList").value
+    let currentScore = this.state.score;
+    
+
+    fetch(`https://nominatim.openstreetmap.org/reverse?lat=${this.state.initialLat}&lon=${this.state.initialLng}&format=geojson`)
+    .then((res) => res.json())
+    .then((json) => {
+      console.log(json.features[0].properties.address.county)
+      if (selectedCounty === json.features[0].properties.address.county) {
+        this.setState({
+          startDisabled: false,
+          guessDisabled: true,
+          quitDisabled: true,
+          town:
+            (json.features[0].properties.address.city ||
+            json.features[0].properties.address.town ||
+            json.features[0].properties.address.village ||
+            json.features[0].properties.address.hamlet),
+          county: json.features[0].properties.address.county,
+          latDisplay: this.state.initialLat,
+          lngDisplay: this.state.initialLng,
+          modal: false,
+          winModal: true
+        })
+
+        alert(`You are correct! Your score is: ${currentScore}`)
+
+      } else {
+        currentScore = currentScore - 10;
+        this.setState({
+          score: currentScore,
+          modal: false
+        })
+        alert(`You are wrong! Your score is now: ${currentScore}`)
+      }  
+    })
+
+  }
+
+  //Function handles a quit, displays the lat/long, town and county
+  handleQuit = (evt) => {
+    evt.preventDefault();
+
+    fetch(
+      `https://nominatim.openstreetmap.org/reverse?lat=${this.state.initialLat}&lon=${this.state.initialLng}&format=geojson`)
+      .then((res) => res.json())
+      .then((json) => {
+        console.log(json)
+        this.setState({
+          startDisabled: false,
+          guessDisabled: true,
+          quitDisabled: true,
+          town:
+            (json.features[0].properties.address.city ||
+            json.features[0].properties.address.town ||
+            json.features[0].properties.address.village ||
+            json.features[0].properties.address.hamlet),
+          county: json.features[0].properties.address.county,
+          latDisplay: this.state.initialLat,
+          lngDisplay: this.state.initialLng,
+        });
+      });
+  
+
+  };
+  
+  //Movement function
+  //Moves the map to the North
+  moveNorth =(evt)=>{
+    evt.preventDefault();
+    let currentLat= this.state.currentLat + 0.002;
+    let currentLng = this.state.currentLng
+    let score= this.state.score -1
+
+
+    this.setState({
+      currentLat: currentLat,
+      score: score, 
+      moveArr: this.state.moveArr.concat([[currentLat, currentLng]])
```

Comment:
This again is bad practice putting `this.state` inside of your `this.setState`. Treating it the same way you did with your score, or by turning this into a callback function would be better. 

This goes the same for each direction button